### PR TITLE
update debugging docs

### PR DESCRIPTION
**Pull request by Mattchewone**
Update references to `ObservableObject` from `DefineMap`.

### CODE REVIEW

File: docs/can-guides/topics/debugging.md [hunk 1]
```diff
@@ -197,14 +198,16 @@ new Observation(function fullName(){
 });
 ```
 
-> **Note:** If your function is a property on an observable map or list like [can-define/map/map],
+> **Note:** If your function is a property on an observable class like [can-observable-object],
 > you don't have to name it. For example, CanJS will name the `fullName` getter in the following example:
```
Comment (phillipskevin) [inline docs/can-guides/topics/debugging.md]: Did you double check that this is true?

File: docs/can-guides/topics/debugging.md [hunk 2]
```diff
@@ -141,24 +141,25 @@ CanJS tries to create useful names to help identify the objects and
...
-name your function by passing `.extend` a name as the first argument:
+If you are using [can-observable-object], name your class:
 
 ```js
-import { DefineMap } from "can";
+import { ObservableObject } from "can";
 
-export default DefineMap.extend("TheNameOfMyType", { /* ... */ })
+export default class TheNameOfMyType extends ObservableObject {
+  static props = { /* ... */ }
```
Comment (phillipskevin) [inline docs/can-guides/topics/debugging.md]: can you add a semicolon `static props = { /* ... */ };`

File: docs/can-guides/topics/debugging.md [hunk 3]
```diff
@@ -197,14 +198,16 @@ new Observation(function fullName(){
...
 > ```js
-> const Person = DefineMap.extend("Person",{
->   fullName: {
->     get: function(){ return this.first + " " + this.last; }
+> class TheNameOfMyType extends ObservableObject {
+>   static props = {
+>     fullName: {
+>       get () { return this.first + " " + this.last; }	
+>     }
 >   }
```
Comment (phillipskevin) [inline docs/can-guides/topics/debugging.md]: semicolon

File: docs/can-guides/topics/debugging.md [hunk 4]
```diff
@@ -345,29 +348,32 @@ Stache templates also have access the [can-stache.helpers.console] methods, maki
...
+	static props = {
+		count: {
+			value({resolve}) {
+				let count = resolve(0);
+				setInterval(() => {
+					resolve(++count);
+				}, 1000);
+			}
+		}
+	}
```
Comment (phillipskevin) [inline docs/can-guides/topics/debugging.md]: semicolon.

File: docs/can-guides/topics/debugging.md [hunk 5]
```diff
@@ -345,29 +348,32 @@ Stache templates also have access the [can-stache.helpers.console] methods, maki
...
-                    resolve(++count);
-                }, 1000);
-            }
-        }
-    }
-});
+class MyCounter extends StacheElement {
+	static view = `
+		{{ console.log(count) }}
+	`
```
Comment (phillipskevin) [inline docs/can-guides/topics/debugging.md]: semicolon.

File: docs/can-guides/topics/debugging.md (hunk 1 above)
Comment (Mattchewone) [reply docs/can-guides/topics/debugging.md]: Using this snippet:
```js
class MyEl extends StacheElement {
  static view = `
  {{ this.myProp }}
  <input value:bind="this.prop">
  `;
  
  static props = {
    prop: {
      type: type.convert(String),
      bind: fromAttribute
    },
    myProp: {
      get: function () {
        return this.prop
      }
    }
  };
}
customElements.define('my-el', MyEl);
```

When setting a breakpoint I see this output:
```
NOTIFY ran task: Observation<MyEl{}'s myProp getter>.onDependencyChange {fn: ƒ, context: my-el, args: Array(2), meta: {…}}
can-log.js:98 DERIVE ran task: Observation<MyEl{}'s myProp getter>.update
```
I'm assuming this is the expected right?

Comment (phillipskevin) [reply docs/can-guides/topics/debugging.md]: Interesting. I assume this is because of the proxy? We should investigate.

**Comment by phillipskevin** [comment]
Will review after publishing.

**Pull request merged**